lib/utils/cli.go

```diff
@@ -196,8 +197,15 @@ func UserMessageFromError(err error) string {
 			}
 			fmt.Fprintln(&buf, AllowNewlines(trace.Unwrap(er).Error()))
 		} else {
-			fmt.Fprintln(&buf, AllowNewlines(err.Error()))
+			strErr := err.Error()
+			// Error can be of type trace.proxyError where error message didn't get captured.
```

Comment:
This is somewhat misleading. Did you have a specific error that did not transport the message over the wire? All errors coming from the server over HTTP will be of type [trace.proxyError](https://github.com/gravitational/trace/blob/bbab4b868a4a8a002ffc76d005d53996506ef059/trace.go#L594) and it's supposed to transport the message with it - if that's not the case here, it's probably a bug to be fixed.

Reply:
I was trying to purposely cause errors when logging in with sso through CLI and I found errors came back as empty strings. I traced into it and I thought maybe it was by design to prevent sensitive error messages from auth from being exposed?

This is how errors get marshalled: https://github.com/gravitational/trace/blob/master/httplib.go#L87
And this is how errors get unmarshalled: https://github.com/gravitational/trace/blob/master/httplib.go#L59
And this is how requests are made in both client and proxy: https://github.com/gravitational/teleport/blob/master/lib/auth/clt.go#L251

Given tsh client -> proxy -> auth make request
First marshal/unmarshal auth to proxy: when error from auth gets marshalled,  the error is of type `TraceErr`, and when it gets unmarshalled, error is unmarshaled into `TraceErr` and wrapped in `proxyErr`

Second marshal/unmarshal proxy to client: When this proxyErr gets marshalled, b/c it is not a type of `TraceErr`, everything gets stripped except error field, and when this gets unmarshalled it gets unmarshalled into trace error struct that only has `message` as its field, so the error field never gets saved and message is empty.

Reply:
Ok, I see. This a bug in trace and should be definitely fixed so it's possible to send errors over the wire multiple times. I'll see how complicated this would be.

Reply:
Since this is a bug, I went back to all console logins `github`, `oidc`, `saml`, in the webapi layer and made sure to return a generic error message for all errors to hide sensitive data. This is what we did for UI dashboard logins (and validating auth callbacks) where any errors encountered we redirected users to a generic error page with direction to check teleports log: https://github.com/gravitational/teleport/pull/6108/commits/50078d3cbbaf743e457a719b93b4ce81b5c203e7